src/StdUtils.sol

```diff
@@ -36,10 +36,24 @@ abstract contract StdUtils {
     }
 
     function bound(uint256 x, uint256 min, uint256 max) internal view virtual returns (uint256 result) {
+        require(min <= max, "StdUtils bound(uint256,uint256,uint256): Max is less than min.");
         result = _bound(x, min, max);
         console2.log("Bound Result", result);
     }
 
+    function boundInt(int256 x, int256 min, int256 max) internal view virtual returns (int256 result) {
+        require(min <= max, "StdUtils boundInt(int256,int256,int256): Max is less than min.");
+
+        uint256 _x = x < 0 ? (INT256_MIN_ABS - ~uint256(x) - 1) : (uint256(x) + INT256_MIN_ABS);
+        uint256 _min = min < 0 ? (INT256_MIN_ABS - ~uint256(min) - 1) : (uint256(min) + INT256_MIN_ABS);
+        uint256 _max = max < 0 ? (INT256_MIN_ABS - ~uint256(max) - 1) : (uint256(max) + INT256_MIN_ABS);
```

Comment:
Mind just adding a comment explaining how this works / linking to a reference? Want to make sure it's clear/understandable to maintainers (including myself) 😅

I think what you're doing is shifting all values from the int domain to the uint domain, running `_bound`, then shifting them back? Which makes sense and is how I'd do it, just not intuitive how the shift works

Reply:
Agree, I'll add some comments about this code.

Reply:
[328f4df](https://github.com/foundry-rs/forge-std/pull/253/commits/328f4dfe531b66876ccafa8be1184b923b04c02d)